distribution/test/itests/test-itests-ddf/src/test/java/ddf/test/itests/catalog/MessageBrokerTest.java

```diff
@@ -0,0 +1,319 @@
+
+    @BeforeExam
+    public void beforeExam() throws Exception {
+        basePort = getBasePort();
+        getAdminConfig().setLogLevels();
+
+        getServiceManager().waitForRequiredApps(REQUIRED_APPS);
+        setupCamelContext();
+        configureRestForGuest();
+        getSecurityPolicy().waitForGuestAuthReady(REST_PATH.getUrl() + "?_wadl");
+        FileUtils.copyInputStreamToFile(AbstractIntegrationTest.getFileContentAsStream(
+                "sdk-example-route.xml",
+                AbstractIntegrationTest.class), Paths.get(ddfHome, "etc", "routes")
+                .resolve("sdk-example-route.xml")
+                .toFile());
+    }
+
+    @Before
+    public void before() throws Exception {
+    }
+
+    @Test
+    public void testDynamicRouting() throws Exception {
+        MockEndpoint endpoint = camelContext.getEndpoint(MOCK_EXAMPLE_TEST_ROUTE,
+                MockEndpoint.class);
+        endpoint.expectedBodiesReceived("test Message");
+        sendMessage(SJMS_EXAMPLE_TEST_QUEUE, "test Message");
+        endpoint.assertIsSatisfied(2000L);
+    }
+
+    @Test
+    public void testUndeliveredMessagesRetry() throws URISyntaxException, InterruptedException {
+        MockEndpoint endpoint = camelContext.getEndpoint(MOCK_UNDELIVERED_TEST_ENDPOINT,
+                MockEndpoint.class);
+        endpoint.expectedBodiesReceived("retry");
+
+        BLOW_UP.set(true);
+        sendMessage(SJMS_UNDELIVERED_TEST_QUEUE, "retry");
+
+        String initialDlqMessageId = verifyDlqIsNotEmpty(TIMEOUT_IN_SECONDS, 10);
+        endpoint.assertIsNotSatisfied();
+
+        BLOW_UP.set(false);
+        resendMessageFromDlq(initialDlqMessageId);
+
+        verifyDlqIsEmpty(TIMEOUT_IN_SECONDS, 10);
+        endpoint.assertIsSatisfied();
+
+    }
+
+    @Test
+    public void testUndeliveredMessagesDelete() throws URISyntaxException, InterruptedException {
+        MockEndpoint endpoint = camelContext.getEndpoint(MOCK_UNDELIVERED_TEST_ENDPOINT,
+                MockEndpoint.class);
+        endpoint.expectedBodiesReceived("delete");
+
+        BLOW_UP.set(true);
+        sendMessage(SJMS_UNDELIVERED_TEST_QUEUE, "delete");
+
+        String initialDlqMessageId = verifyDlqIsNotEmpty(TIMEOUT_IN_SECONDS, 10);
+        endpoint.assertIsNotSatisfied();
+
+        BLOW_UP.set(false);
+        deleteMessageFromDlq(initialDlqMessageId);
+
+        verifyDlqIsEmpty(TIMEOUT_IN_SECONDS, 10);
+        endpoint.assertIsNotSatisfied();
+
+    }
+
+    private String verifyDlqIsNotEmpty(long timeout, long pollingInterval) {
+        WaitCondition waitCondition = expect(
+                "The Dead Letter Queue is not empty and contains an undelivered message.").within(
+                timeout,
+                TimeUnit.SECONDS)
+                .checkEvery(pollingInterval, TimeUnit.SECONDS)
+                .until(() -> {
+                    // Get undelivered messages from the DLQ
+                    List<Object> responseMap = getMessagesFromDlq();
```

Comment:
The objects in the list are are Maps, but I agree we should probably still rename this variable.